tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNetCore/BlockingMiddleware.cs

```diff
@@ -0,0 +1,130 @@
+﻿// <copyright file="BlockingMiddleware.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+#if !NETFRAMEWORK
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Datadog.Trace.AppSec;
+using Datadog.Trace.AppSec.Coordinator;
+using Microsoft.AspNetCore.Http;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AspNetCore;
+
+internal class BlockingMiddleware
+{
+    private readonly bool _endPipeline;
+    private readonly RequestDelegate? _next;
+
+    internal BlockingMiddleware(RequestDelegate? next = null, bool endPipeline = false)
```

Comment:
Just a heads up, if we _do_ add support for ASP.NET Core on .NET Framework, we can't directly reference `RequestDelegate`, so this would need to be written. There's no commitment to that yet though, just pointing it out!

Reply:
good to know!, you meant duck typing `RequestDelegate`? I'll add it as a comment in here